cpp_coveralls/gitrepo.py

```diff
@@ -30,6 +30,14 @@ def gitrepo(cwd):
     """
     repo = Repository(cwd)
 
+    try:
+        branch = os.environ['TRAVIS_BRANCH']
+    except KeyError:
+        try:
+            branch = os.environ['APPVEYOR_REPO_BRANCH']
+        except KeyError:
+            branch = repo.git('rev-parse', '--abbrev-ref', 'HEAD').strip()
+
```

Comment:
Sure. But note this will invoke the git subprocess even when either environment variable is set.

I haven't tried to support it yet (for now I'm using git shallow clones), but AppVeyor can actually get the source from github without doing a git clone.  And I believe that in that case one needs to get all the info from the [environment variables](https://www.appveyor.com/docs/environment-variables), particularly all `APPVEYOR_REPO_COMMIT_*` ones.  Any attempt to get the info through git command will spew error messages like `fatal: Not a git repository (or any of the parent directories): .git`.

I don't like the nested try... except much either myself, but it was the best way I could come up that avoids the git subprocess.

Anyway, I don't care much either way now, since I'm using a git shallow clone, since it's necessary for all other fields, and I'm not sure it's worth trying to support AppVeyor clone-less mode.